main.tf

```diff
@@ -1,3 +1,336 @@
+    value: kubernetes.default.svc.cluster.local
+  %{endif}
+EOF
+
+  eso_helm_release_values_workerselector = var.eso_cluster_nodes_configuration == null ? "" : <<-EOF
+nodeSelector: { ${var.eso_cluster_nodes_configuration.nodeSelector.label}: ${var.eso_cluster_nodes_configuration.nodeSelector.value} }
+tolerations:
+- key: ${var.eso_cluster_nodes_configuration.tolerations.key}
+  operator: ${var.eso_cluster_nodes_configuration.tolerations.operator}
+  value: ${var.eso_cluster_nodes_configuration.tolerations.value}
+  effect: ${var.eso_cluster_nodes_configuration.tolerations.effect}
+webhook:
+  nodeSelector: { ${var.eso_cluster_nodes_configuration.nodeSelector.label}: ${var.eso_cluster_nodes_configuration.nodeSelector.value} }
+  tolerations:
+  - key: ${var.eso_cluster_nodes_configuration.tolerations.key}
+    operator: ${var.eso_cluster_nodes_configuration.tolerations.operator}
+    value: ${var.eso_cluster_nodes_configuration.tolerations.value}
+    effect: ${var.eso_cluster_nodes_configuration.tolerations.effect}
+certController:
+  nodeSelector: { ${var.eso_cluster_nodes_configuration.nodeSelector.label}: ${var.eso_cluster_nodes_configuration.nodeSelector.value} }
+  tolerations:
+  - key: ${var.eso_cluster_nodes_configuration.tolerations.key}
+    operator: ${var.eso_cluster_nodes_configuration.tolerations.operator}
+    value: ${var.eso_cluster_nodes_configuration.tolerations.value}
+    effect: ${var.eso_cluster_nodes_configuration.tolerations.effect}
+EOF
+}
+
+resource "helm_release" "external_secrets_operator" {
+  depends_on = [module.eso_namespace, data.kubernetes_namespace.existing_eso_namespace]
+
+  name      = "external-secrets"
+  namespace = local.eso_namespace
+  chart     = "oci://icr.io/ibm-iac-charts/external-secrets"
+  version   = "0.10.5"
+  wait      = true
+
+  set {
+    name  = "image.repository"
+    type  = "string"
+    value = local.eso_image_repo
+  }
+
+  set {
+    name  = "image.tag"
+    type  = "string"
+    value = local.eso_image_tag_digest
+  }
+
+  set {
+    name  = "webhook.image.repository"
+    type  = "string"
+    value = local.eso_image_repo
+  }
+
+  set {
+    name  = "webhook.image.tag"
+    type  = "string"
+    value = local.eso_image_tag_digest
+  }
+
+  set {
+    name  = "certController.image.repository"
+    type  = "string"
+    value = local.eso_image_repo
+  }
+
+  set {
+    name  = "certController.image.tag"
+    type  = "string"
+    value = local.eso_image_tag_digest
+  }
+
+  # The following mounts are needed for the CRI based authentication with Trusted Profiles
+  values = [local.eso_helm_release_values_cri, local.eso_helm_release_values_workerselector]
+}
+
+resource "helm_release" "pod_reloader" {
+  depends_on = [module.eso_namespace, data.kubernetes_namespace.existing_eso_namespace]
+  count      = var.reloader_deployed == true ? 1 : 0
+  name       = "reloader"
+  namespace  = local.eso_namespace
+  chart      = "oci://icr.io/ibm-iac-charts/reloader"
```

Comment:
same as https://github.com/terraform-ibm-modules/terraform-ibm-external-secrets-operator/pull/6/files#r1862494213 - pull from the `https://stakater.github.io/stakater-charts` registry